src/python-frontend/python_converter.cpp

```diff
@@ -518,26 +519,37 @@ function_id python_converter::build_function_id(const nlohmann::json &element)
   bool is_member_function_call = false;
   const nlohmann::json &func_json = element["func"];
   const std::string &func_type = func_json["_type"];
-  std::string func_name, obj_name;
+  std::string func_name, obj_name, class_name;
   std::string func_symbol_id = create_symbol_id();
 
   if (func_type == "Name")
     func_name = func_json["id"];
-  else if (func_type == "Attribute")
+  else if (func_type == "Attribute") // Handling obj_name.func_name() calls
   {
+    is_member_function_call = true;
     func_name = func_json["attr"];
     obj_name = func_json["value"]["id"];
-    if (json_utils::is_module(obj_name, ast_json))
+    if (
+      !is_class(obj_name, ast_json) &&
+      json_utils::is_module(obj_name, ast_json))
+    {
       func_symbol_id = create_symbol_id(imported_modules[obj_name]);
-    else
-      is_member_function_call = true;
+      is_member_function_call = false;
+    }
   }
 
+  // build symbol_id
   if (func_name == "len")
   {
     func_name = __ESBMC_get_object_size;
     func_symbol_id = "c:@F@" + __ESBMC_get_object_size;
   }
+  else if (is_builtin_type(obj_name))
+  {
+    class_name = obj_name;
+    func_symbol_id =
```

Comment:
Shall we consider using a `std::stringstream` for building the func_symbol_id string to be more efficient and readable?